sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp

```diff
@@ -10,7 +10,8 @@
 #include "../src/private/keyvault_protocol.hpp"
```

Comment:
This is an scape. Public headers shall not include private headers.  Make the required changes to only include the private headers from within the cpp files

Reply:
can't do that here , i need the protocol header for the definition of m_protocol member of the client class